test/fail_compilation/nosharedaccess.d

```diff
@@ -0,0 +1,31 @@
+/* REQUIRED_ARGS: -preview=nosharedaccess
+ * TEST_OUTPUT:
+---
+fail_compilation/nosharedaccess.d(1010): Error: direct access to shared `j` is not allowed, see `core.atomic`
+fail_compilation/nosharedaccess.d(1011): Error: direct access to shared `j` is not allowed, see `core.atomic`
+fail_compilation/nosharedaccess.d(1012): Error: direct access to shared `*p` is not allowed, see `core.atomic`
+fail_compilation/nosharedaccess.d(1013): Error: direct access to shared `a[0]` is not allowed, see `core.atomic`
+fail_compilation/nosharedaccess.d(1014): Error: direct access to shared `s.si` is not allowed, see `core.atomic`
+fail_compilation/nosharedaccess.d(1015): Error: direct access to shared `t` is not allowed, see `core.atomic`
+fail_compilation/nosharedaccess.d(1015): Error: direct access to shared `t.i` is not allowed, see `core.atomic`
+---
+*/
+
+#line 1000
+
+struct S
+{
+    shared(int) si;
+    int i;
+}
+
+void func(shared int j, shared(int)* p, shared(int)[] a, ref S s, ref shared S t)
+{
+    int i;
+    j = i;
+    i = j;
+    i = *p;
+    i = a[0];
+    i = s.si;
+    i = t.i;
```

Comment:
Should the test prove that writes are denied too?
I feel like it might also be good to prove that taking the address is okay, and passing by pointer/ref.

Reply:
line 25

Reply:
Sorry, I missed that one hiding in there.

Reply:
I suppose those other tests are implied by the function arguments.

Reply:
done

Reply:
please assert that passing the shared to non shared parameter fails like it is supposed to.